stabilizer/src/main/java/com/hazelcast/stabilizer/provisioner/Provisioner.java

```diff
@@ -343,6 +345,40 @@ public void download(String dir) {
         echoImportant("Finished Downloading Artifacts of %s machines", addresses.size());
     }
 
+    public List<String> getPublicAdresses(List<AgentAddress> addresses){
+        List<String> result = new LinkedList<String>();
+        for(int i = 0; i < addresses.size(); i++){
+            result.add(addresses.get(i).publicAddress);
+        }
+        return result;
+    }
+
+    public void upload() throws IOException {
+        if(uploadPath != null){
+            log.info(format("Copying file '%s' to agents", uploadPath));
+            List<File> upload = Utils.getFilesFromClassPath(uploadPath);
+            List<String> publicAddresses = getPublicAdresses(addresses);
+            for(String ip : publicAddresses){
+                for (File file : upload){
+                    String syncCommand =
+                            format("rsync --ignore-existing -av -e \"ssh %s\" %s %s@%s:hazelcast-stabilizer-%s/workers/resources/",
+                                    props.get("SSH_OPTIONS", ""),
+                                    file.getAbsolutePath(),
+                                    props.get("USER"),
+                                    ip,
+                                    getVersion()
+                            );
+                    bash.execute(syncCommand);
+                }
+                log.info("    " + ip + " copied");
+            }
+            log.info(format("Finished copying file path '%s' to agents", uploadPath));
+        }
+        else{
+            log.info("Could not upload files");
```

Comment:
If the user has not specified any files to upload, he should not get an info message but a debug message because apparently he doesn't care for it so you should not inform him about things he doesn't care for.